File: 199-BinaryTreeRightSideView/199-BinaryTreeRightSideView.py
# Last updated: 08/05/2026, 09:31:07
# Definition for a binary tree node.
import collections
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        ans=[]
        
        q= collections.deque([root])

        while(q):
            rightside=None
            qlen= len(q)
            
            for i in range(qlen):

                node = q.popleft()
                if node:
                    q.append(node.left)

                    q.append(node.right)

                    rightside= node

            if rightside is not None:
                ans.append(rightside.val)

        return ans        

File: 199-BinaryTreeRightSideView/199-BinaryTreeRightSideView.cpp
// Last updated: 08/05/2026, 09:31:02
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
 #include<deque>
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        deque<TreeNode*> q;
        q.push_back(root);
        TreeNode* rightside;
        int qlen;
        vector<int> ans;
        while(!q.empty()){
            rightside=nullptr;
            qlen= q.size();
            for(int i=0; i<qlen;i++){
                TreeNode* node=q.front();
                q.pop_front();
                if(node){
                    
                if (node->left) q.push_back(node->left);
                if (node->right) q.push_back(node->right);
                    rightside= node;}
            }
            if(rightside){
                ans.push_back(rightside->val);
            }
        }

        return ans;


    }
};